core/tests/README.md

```diff
@@ -0,0 +1 @@
+The root-files-config.json holds the root files that are valid; a root file is a file which other files don't depend on. There are two places a root file can go, the RUN_NO_TESTS_ROOT_FILES and RUN_ALL_TESTS_ROOT_FILES. If the CI gives you an error regarding this config file, please add the root file to one of these lists, depending on if it needs to run tests or not. For more information, refer to the wiki here: https://github.com/oppia/oppia/wiki/Partial-CI-Tests-Structure#root-files-config
```

Comment:
I see that you added a technical definition of a root file, and while that's helpful, I still think readers are going to be confused because the notion of files depending on each other is somewhat vague without the extra context that we're trying to decide which tests need to be run. Here's what I'd recommend:

> To decide which CI tests need to be run on a PR, we need to know how files depend on each other. These dependencies can be described as a set of dependency trees where the files depend on their parents and are dependencies of their children. The root of each tree is therefore a file that no other file depend upon. These root files are stored in root-files-config.json under one of two keys: RUN_NO_TESTS_ROOT_FILES for files that don't affect any functionality and RUN_ALL_TESTS_ROOT_FILES for files that could change Oppia's behavior. If the CI gives you an error regarding this config file, please add the root file to one of these lists, depending on if it needs to run tests or not. For more information, refer to the wiki here: https://github.com/oppia/oppia/wiki/Partial-CI-Tests-Structure#root-files-config

Reply:
Thanks, your suggestion looks good, so I added it.